### DISCUSSION TITLE
Attachments URL + authentication

### DISCUSSION THREAD

**Discussion by man-tax** [Budibase/budibase#5514] [Help] [answered]
Hi everyone,

as it was mentioned in #3579, attachments being publicly accessible is a bug and supposed to be fixed. I'm curious as to when, as my files are still accessible by anyone. Or did I miss where to set permissions?

Also I'm having issues with the links to the attachments. I'm using the repeater block (data {{ repeater.table.attachment }})  for displaying links ({{ repeaterblock.attachment.url }}) to the attachments. It generates the syntax 

...myserver/app/start#/prod-budi-app-assets/app_4a97fc30ea25466..., but they actually sit at 
...myserver/prod-budi-app-assets/app_4a97fc30ea25466... 

This makes them not accessible via my links. The links in the table builder, and if I place a table on my screens for example, work fine.

Thanks a lot!
Sebastian

**Comment by mjashanks** [comment, marked as answer]
Hi Sebastian.

I will followup on this one.

Regarding your binding issue. The URL returned from the binding is a relative one `/prod-budi-app-assets/app_4a97fc30ea25466...` meaning that your browser will automatically prepend the current domain or path to it.

In order to get this to work, your binding will need to be `https://myserver{{ repeaterblock.attachment.0.url }}` (I also added `.0.url` in there as I think you need that too).

**Reply by man-tax** [reply]
Thanks, Michael! Perfectly working now (without the .0. though, but will keep in mind)

Any chance you can share some information on the authentication for attachments? I was hoping I've just missed something. I managed to set permissions for the bucket in the Minio console to private, but as I was expecting, this makes them not accessible anymore at all (so not even from inside the table builder). Thanks!

**Reply by mjashanks** [reply]
Yea - unfortunately, this slipped through the cracks; we haven't addressed it. We are going to try and get it scheduled for the next few months - so should be by the end of June.

**Reply by man-tax** [reply]
Thanks for the transparency! Knowing this will come is fine for the moment.

In the meantime: I did manage to change the minio policy so it allows access to requests from inside my app only using the aws:referer condition. This doesn't give 100% security of course, but an additional layer I think. Also I've just managed to set up my S3 bucket as a data source. My server is with AWS also, but I don't think this provides much advantage over minio in terms of permissions, does it? I'll see.

**Reply by aowdnmp** [reply]
Still not addressed? I was trying to access an uploaded file URL and seems still publicly accessible. Am I missing something? Thanks

**Reply by mike12345567** [reply]
> Still not addressed? I was trying to access an uploaded file URL and seems still publicly accessible. Am I missing something? Thanks

Hi @aowdnmp - the URLs returned with Budibase are pre-signed URLs from S3/MinIO depending on the environment - these will be publicly accessible while the URL is valid, but a different URL will be generated each time the attachments are fetched. The attachments are no longer publicly indexable in anyway.

**Reply by aowdnmp** [reply]
In my case, file is stored inside its own budibase db.
The link is kind of https://.....p.cloudfront.net/......... and seems not changing over time avene if I accessed it several times. Copying the link I can access the document from an in private browser session. Is there some known timers on this link expiration? The link I saved from several 4+ hrs ago seems still working publicly.
I'm new to budibase so I'm just experimenting. Thanks.

**Reply by mike12345567** [reply]
Hi @aowdnmp yes - it will keep using the same link for a session and that link will be valid in any browser for 24 hours.

**Reply by aowdnmp** [reply]
Ok, understood, so I will test it again tomorrow, now it's clear, thanks!

**Reply by NexusEFR** [reply]
Hello, is it possible to extend the expiration of a link to an attachment created by a form, which sends this link in an email but becomes invisible after 1 hour? In a self-hosted installation using Docker under Unraid (all-in-one), the variable MINIO_PRESIGNED_URL_EXPIRY= 604800 doesn't seem to do anything. Is there any parameter that can be configured and maintained after updates so that these links inserted in the email work for up to 7 days? Thanks for the help.
